server/lib/src/server/migrations.rs

```diff
@@ -294,6 +302,75 @@ impl QueryServerWriteTransaction<'_> {
         }
     }
 
+    /// Migration domain level 7 to 8 (1.4.0)
+    #[instrument(level = "info", skip_all)]
+    pub(crate) fn migrate_domain_7_to_8(&mut self) -> Result<(), OperationError> {
```

Comment:
Not really for this one but I wonder if we can't codegen these methods via a macro rule. would take a little bit of the repetition out especially given that all the methods within the module have the same name

Reply:
Yes, we could, but then if we change one migration we have to then duplicate the macro again, and it has it's own issues. I think for now we do it the correct way, then we simplify, then optimise. It may be stupid, but it's correct.